request/consumer.go

```diff
@@ -0,0 +1,49 @@
+package request
+
+import (
+	"github.com/ONSdigital/dp-observation-extractor/kafka"
+	"github.com/ONSdigital/dp-observation-extractor/schema"
+	"github.com/ONSdigital/go-ns/log"
+)
+
+// MessageConsumer provides a generic interface for consuming []byte messages
+type MessageConsumer interface {
+	Incoming() chan kafka.Message
+	Closer() chan bool
+}
+
+// Handler represents a handler for processing a single request.
+type Handler interface {
+	Handle(request *Request) error
+}
+
+// Consume convert them to request instances, and pass the request to the provided handler.
+func Consume(messageConsumer MessageConsumer, handler Handler) {
+	for message := range messageConsumer.Incoming() {
+
+		request, err := Unmarshal(message)
```

Comment:
A good point. I had originally had a generic concept of request to get the service to do something. I can see how this can be misinterpreted, or how inconsistent log message can become confusing if they are all referring to events. I have updated the code to use events instead of requests throughout.